share/shwap/p2p/bitswap/getter.go

```diff
@@ -112,36 +111,34 @@ func (g *Getter) GetShares(
 
 	err := Fetch(ctx, g.exchange, hdr.DAH, blks, WithStore(g.bstore), WithFetcher(ses))
 	if err != nil {
+		// handle partial fetches
+		var fetched int
+		smpls := make([]shwap.Sample, len(blks))
+		for i, blk := range blks {
+			if smpl := blk.(*SampleBlock).Container; !smpl.IsEmpty() {
+				smpls[i] = smpl
+				fetched++
+			}
+		}
+
 		span.RecordError(err)
 		span.SetStatus(codes.Error, "Fetch")
+		if fetched > 0 {
+			span.SetAttributes(attribute.Int("fetched", fetched))
+			return smpls, err
+		}
 		return nil, err
 	}
 
-	shares := make([]libshare.Share, len(blks))
+	smpls := make([]shwap.Sample, len(blks))
 	for i, blk := range blks {
-		shares[i] = blk.(*SampleBlock).Container.Share
+		c := blk.(*SampleBlock).Container
+		c.Proof = &nmt.Proof{} // TODO
```

Comment:
Without this line `TestPruneAll` & `TestPrunePartialFailed` will fail 'cause `smpl.IsEmpty()` will be true. Not sure how to fetch `.Proof` properly here, @Wondertan any ideas?

CC: @vgonkivs

Reply:
Looking into it now.

We definitely should find a different option, as this one basically erases the part of the response users may want.

Reply:
Ok, there is a bug in the test itself. The blocks are always coming empty.

cc @walldiss.

Reply:
Dunno how that worked previously, like it was succeeding for some other reason, like a false positive

Reply:
Almost finished the fix

Reply:
e13f2c24